source/clear-linux/guides/maintenance/kernel-modules-dkms.rst

```diff
@@ -0,0 +1,307 @@
+document.
+
+:abbr:`DKMS (Dynamic Kernel Module System)` is a framework that facilitates
+the building and installation of kernel modules. This allows |CL| to provide
+hooks that automatically rebuild modules against new kernel versions. 
+
+
+.. contents:: :local:
+   :depth: 1
+   :backlinks: top
+
+
+
+.. include:: kernel-modules.rst
+   :start-after: kernel-modules-availability-begin:
+   :end-before: kernel-modules-availability-end:
+
+
+
+
+Install DKMS on |CL|
+====================
+
+The *kernel-native-dkms* bundle provides the :command:`dkms`: program and
+Linux kernel headers, which are required for compiling kernel modules.
+
+The *kernel-native-dkms* bundle also:
+
+* Adds a systemd update trigger
+  (:file:`/usr/lib/systemd/system/dkms-new-kernel.service`) to automatically
+  run DKMS to rebuild modules after a kernel upgrade occurs with :ref:`swupd
+  update <swupd-guide>`.
+
+* Disables kernel modules signature verification by appending a kernel
+  command-line parameter (:command:`module.sig_unenforce`) from
+  :file:`/usr/share/kernel/cmdline.d/clr-ignore-mod-sig.conf`.
+
+* Adds a notification to the Message of the Day (MOTD) indicating kernel
+  modules signature verification is disabled. 
+
+.. warning::
+
+   #. It is important to always review the output of :command:`swupd update` to
+      make sure kernel modules rebuilt against the new kernel successfully. This is
+      especially important for systems where a successful boot relies on a kernel
+      module.
+
+   #. Kernel modules installed with DKMS typically put source and binary files
+      under the :file:`/usr` subdirectory. This conflicts with the :ref:`stateless
+      <stateless>` design principles of |CL| and the :command:`swupd verify
+      --fix` command should be avoided with out-of-tree kernel modules in use.   
+
+Install the *kernel-native-dkms* bundle:
+
+#. From a |CL| system, ensure you are running the *native* kernel. 
+   Currently only the native kernel is enabled to build and load
+   out-of-tree modules.
+
+   .. code-block:: bash
+
+      $ uname -r
+      5.XX.YY-ZZZZ.native
+
+   Ensure *.native* is in the kernel name
+
+#. Install the `kernel-native-dkms`.
+
+   .. code-block:: bash
+
+      sudo swupd bundle-add kernel-native-dkms
+   
+
+#. Update the |CL| bootloader and reboot.
+
+   .. code-block:: bash
+
+      sudo clr-boot-manager update 
+      reboot
+
+
+
+Build, install, and load an out-of-tree module
+==============================================
+In some cases you may need an out-of-tree kernel module that is not available
+through |CL|.
+
+
+Prerequisites 
+-------------
+
+You can build and load out-of-tree kernel modules, however you must:
+
+* Disable Secure Boot in UEFI/BIOS. The loading of new out-of-tree modules
+  modifies the signatures Secure Boot relies on for trust. 
+  
+* Have a kernel module package in the form of source code and/or precompiled
```

Comment:
precompiled binaries still won't work -- even though we disable module signatures, we keep the version checks that reject any binary not built against the same source tree.